tools/pkglint/lib/rules.ts

```diff
@@ -1442,6 +1442,28 @@ export class ConstructsDependency extends ValidationRule {
   }
 }
 
+/**
+ * Packages must depend on 'assert-internal', not on '@aws-cdk/assert'
+ */
+export class AssertDependency extends ValidationRule {
+  public readonly name = 'assert/assert-dependency';
+
+  public validate(pkg: PackageJson) {
+    const devDeps = pkg.json.devDependencies ?? {};
+
+    if ('@aws-cdk/assert' in devDeps) {
```

Comment:
Minor: should we check for both devDeps and deps to be safe?

Reply:
It shouldn't be in regular deps to begin with.